packages/components/src/buttons/src/Button.module.css

```diff
@@ -0,0 +1,277 @@
+    --border: var(--hop-button-component-tertiary-border);
+}
+
+.hop-button--tertiary[data-disabled]:not([data-loading]) {
+    --background-color: var(--hop-button-component-tertiary-background-color-disabled);
+    --color: var(--hop-button-component-tertiary-color-disabled);
+    --border: var(--hop-button-component-tertiary-border-disabled);
+}
+
+.hop-button--tertiary[data-hovered] {
+    --background-color: var(--hop-button-component-tertiary-background-color-hover);
+    --color: var(--hop-button-component-tertiary-color-hover);
+    --border: var(--hop-button-component-tertiary-border-hover);
+}
+
+.hop-button--tertiary[data-pressed] {
+    --background-color: var(--hop-button-component-tertiary-background-color-pressed);
+    --color: var(--hop-button-component-tertiary-color-pressed);
+    --border: var(--hop-button-component-tertiary-border-pressed);
+}
+
+.hop-button--upsell {
+    --background-color: var(--hop-button-component-upsell-background-color);
+    --color: var(--hop-button-component-upsell-color);
+    --border: var(--hop-button-component-upsell-border);
+}
+
+.hop-button--upsell[data-disabled]:not([data-loading]) {
+    --background-color: var(--hop-button-component-upsell-background-color-disabled);
+    --color: var(--hop-button-component-upsell-color-disabled);
+    --border: var(--hop-button-component-upsell-border-disabled);
+}
+
+.hop-button--upsell[data-hovered] {
+    --background-color: var(--hop-button-component-upsell-background-color-hover);
+    --color: var(--hop-button-component-upsell-color-hover);
+    --border: var(--hop-button-component-upsell-border-hover);
+}
+
+.hop-button--upsell[data-pressed] {
+    --background-color: var(--hop-button-component-upsell-background-color-pressed);
+    --color: var(--hop-button-component-upsell-color-pressed);
+    --border: var(--hop-button-component-upsell-border-pressed);
+}
+
+.hop-button--negative {
+    --background-color: var(--hop-button-component-negative-background-color);
+    --color: var(--hop-button-component-negative-color);
+    --border: var(--hop-button-component-negative-border);
+}
+
+.hop-button--negative[data-disabled]:not([data-loading]) {
+    --background-color: var(--hop-button-component-negative-background-color-disabled);
+    --color: var(--hop-button-component-negative-color-disabled);
+    --border: var(--hop-button-component-negative-border-disabled);
+}
+
+.hop-button--negative[data-hovered] {
+    --background-color: var(--hop-button-component-negative-background-color-hover);
+    --color: var(--hop-button-component-negative-color-hover);
+    --border: var(--hop-button-component-negative-border-hover);
+}
+
+.hop-button--negative[data-pressed] {
+    --background-color: var(--hop-button-component-negative-background-color-pressed);
+    --color: var(--hop-button-component-negative-color-pressed);
+    --border: var(--hop-button-component-negative-border-pressed);
+}
+
+.hop-button__Spinner {
```

Comment:
Should we stylize these? I though they were escape hashes for devs?

Reply:
can you elaborate @francis? this is not the global css class for spinner, this is a .hop-button__PART selector we forward to the spinner